plugins/flytekit-pydantic/tests/test_type_transformer.py

```diff
@@ -0,0 +1,167 @@
+import os
+from typing import Any, Dict, List, Optional, Type, Union
+
+import flytekitplugins.pydantic  # noqa F401
+import pytest
+from flytekitplugins.pydantic import BaseModelTransformer
+from pydantic import BaseModel, Extra
+
+from flytekit import task, workflow
+from flytekit.core.type_engine import TypeTransformerFailedError
+from flytekit.types import directory
+from flytekit.types.file import file
+
+
+class TrainConfig(BaseModel):
+    """Config BaseModel for testing purposes."""
+
+    batch_size: int = 32
+    lr: float = 1e-3
+    loss: str = "cross_entropy"
+
+    class Config:
+        extra = Extra.forbid
+
+
+class Config(BaseModel):
+    """Config BaseModel for testing purposes with an optional type hint."""
+
+    model_config: Optional[Union[Dict[str, TrainConfig], TrainConfig]] = TrainConfig()
+
+
+class ConfigRequired(BaseModel):
+    """Config BaseModel for testing purposes with required attribute."""
+
+    model_config: Union[Dict[str, TrainConfig], TrainConfig]
+
+
+class ConfigWithFlyteFiles(BaseModel):
+    """Config BaseModel for testing purposes with flytekit.files.FlyteFile type hint."""
+
+    flytefiles: List[file.FlyteFile]
+
+
+class ConfigWithFlyteDirs(BaseModel):
+    """Config BaseModel for testing purposes with flytekit.files.FlyteFile type hint."""
+
+    flytedirs: List[directory.FlyteDirectory]
+
+
+class ChildConfig(Config):
+    """Child class config BaseModel for testing purposes."""
+
+    d: List[int] = [1, 2, 3]
+
+
+@pytest.mark.parametrize(
```

Comment:
I think it would be nice to add an example of `ConfigWithFlyteFiles` and `ConfigWithFlyteDirs` to this roundtrip test.